### PR TITLE
[IMP] website_sale: review product page breadcrumb

### PR DESCRIPTION
**Pull request by mano-odoo**
This commit aims to improve the navigation consistency between /shop,
subcategories, and /product page.

Product page:
- Swapping breadcrumb and searchbar for consistency
- Review spacing

Minor fix on shop page:
- "Sort by" inside the filter dropdown

task-3986938
part of task-3986921

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by mano-odoo** [comment]
@dalaOdoo feel free to comment your testings results / remarks on this PR, I temporarily cherry picked the "soon to be merged" commit on this PR for you to test with the full features 😄 (I will drop them once you're done)

**Comment by dalaOdoo** [comment]
@mano-odoo LGTM

File: addons/website_sale/views/templates.xml [hunk 1]
```diff
@@ -794,7 +863,10 @@
             <t t-set="entries" t-value="categories"/>
         </t>
 
-        <div t-if="entries" class="o_wsale_filmstip_container d-flex align-items-stretch mb-2 overflow-hidden">
+        <div
+            t-if="entries"
+            class="o_wsale_filmstip_container d-flex align-items-stretch overflow-hidden"
+        >
```
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:869]: I know that it's the expected behaviour according to the design @paju-odoo validated, but I'm still not sure about the visual result, sorry 😕 
I see at least 3 issues:
- By default, the "fullscreen" tiny scrollbar resembles too much an horizontal separator. The effect it's even more prominent if your OS's scrollbar design doesn’t match.

- If a sidebar is available (normally it's the case) I'd like to visually prioritize that rather than the filmstrip. The upper left corner it's way too important since it's the first zone that the user will "scan" with his eyes.

|  |
|--------|
| image | 


- When you have just one category (or a few) the space it's not well optimised. Somehow it was the case even before, but in this PR the issue affects the entire page 😕 (see master screenshoot for comparison).

|  |
|--------|
|  this PR image |
|  master image |
 
In conclusion I'd keep the filmistrip where it is .
I'm well aware that some users may find this layout visually interesting, but eventually it should be an option; not default IMO

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Hei @mano-odoo ,  the overall navigation it's greatly improved, well done 👍 
Some nitpicking comments for you + one regarding the overall visual result.

Thanks 🙌

File: addons/website_sale/views/templates.xml (hunk 1 above)
Comment (dalaOdoo) [reply addons/website_sale/views/templates.xml:869]: @stefanorigano 
1. is it really worse than today ? I think it was a bit the idea of moving categories above, separate ir more clearly. Can't we simply improve the layout ? 
2. Not sure user with few number of categories will use this feature. If you have a few number of categories it's probably better to add the filter on the left don't you think ?

Comment (dalaOdoo) [reply addons/website_sale/views/templates.xml:869]: @stefanorigano I navigated on several eshop and the few cases I found with a filter on categories on top of the shop are always above everything.

Comment (dalaOdoo) [reply addons/website_sale/views/templates.xml:869]: I agree to keep the above one instead of the in grid.

**Comment by mano-odoo** [comment]
Note: I added a commit to fix the "Sort by" label near the dropdown. Which should not move as much as we had in master since the layout changed, but could still be misplaced in case of long pricelist/dropdown entry

Now: 
image

File: addons/website_sale/views/templates.xml (hunk 1 above)
Comment (stefanorigano) [reply addons/website_sale/views/templates.xml:869]: @dalaOdoo 

> 1 is it really worse than today ? 

As said, some people would love it, but others don't. FP for example, things that "master is good / better" and I do agree with him on this. Let's not impose debatable choices, we have editor options to handle such scenario

> 2 Not sure user with few number of categories will use this feature. If you have a few number of categories it's probably better to add the filter on the left don't you think ?

Nope. You may have dozens of categories, but as soon as you "navigate" them you'll end-up in a scenario with just one sub-category (or a few)

@mano-odoo 

> ... But there is room for improvement A solution such as the following on nested categories,...

Nice, I don't see such solution in your PR 😉 ... I don't think that we can ask people to wait till 19.0 to got it fixed.

Comment (dalaOdoo) [reply addons/website_sale/views/templates.xml:868]: let's move forward then

Comment (mano-odoo) [reply addons/website_sale/views/templates.xml:868]: Hello @livaios, I would say it makes more sense to keep the sort by and the viewswitcher closer to the grid because it interacts directly with it. 

Filtering or changing the view in list is closely related to the grid while the search and the filmstrip are navigation elements. They serve a different purpose. It was more clear with the borders like in the initial mockup though: 
image


Note in the end after a whole lot of discussion, FP prefers what we have in master for the /shop page. I'll update this PR with only the minor fixes (sort by dropdown) and the navigation fixes on the /product page.

**Comment by mano-odoo** [comment]
@livaios I'll just ask a design confirmation tomorrow before merging but you can already have a look at the code. Thanks a lot 🙇

**Review by livaios — CHANGES_REQUESTED** [review]
A few nitpicking comments but other than that LGTM

File: addons/website_sale/views/templates.xml [hunk 2]
```diff
@@ -1190,37 +1190,56 @@
...
+                                    <a
+                                        class="py-2 py-lg-0"
+                                        t-att-href="keep('/shop/category/%s' % slug(category), category=0, attribute_value=0, tags=0)"
+                                    >
+                                        <i class="oi oi-chevron-left d-lg-none me-1" role="presentation"/><t t-out="category.name"/>
+                                    </a>
+                                </li>
+                                <li t-else="" class="o_not_editable breadcrumb-item d-lg-none">
+                                    <a class="py-2 py-lg-0" t-att-href="keep(category=0, attribute_value=0, tags=0)">
+                                        <i class="oi oi-chevron-left me-1" role="presentation"/>All Products
```
[the remark this answers is not in the document]
Comment (mano-odoo) [reply addons/website_sale/views/templates.xml:1220]: I avoid going to the line to avoid whitespace which would create this effect 😄 
allproduct

File: addons/website_sale/static/src/scss/website_sale.scss [hunk 3]
```diff
@@ -204,6 +204,16 @@ $input-border-color: $gray-400;
             max-width: $o-wsale-dropdown-width-overflow;
         }
     }
+
+    @include media-breakpoint-down(lg) {
+        .o_wsale_breadcrumb .breadcrumb-item {
+            padding-left: 0;
+
+            &::before {
+                content: "";
```
Comment (stefanorigano) [inline addons/website_sale/static/src/scss/website_sale.scss:213]: This kind of "workarounds" should come with comment😉

**Review by stefanorigano — APPROVED** [review]
Thanks @mano-odoo , UI is clean and the UX it's improved.
One nitpicking remark + you should undraft the PR since it's under tech. review already😉

Also, about the commits message:
`[IMP] website_sale: product page`, and
`[IMP] website_sale: sort by inside the dropdown`
are not really self-explanatory. People should be able to understand what's going on just reading at the title 😉   Also `sort by` -> `"sort-by"` or `[sort-by]`

Thanks again, well done 👍

**Comment by chevalierv** [comment]
Hello @mano-odoo 

Good job on this ! 

@robodoo r+

Have a nice day !

**Comment by chevalierv** [comment]
@rodoboo rebase-ff

**Comment by chevalierv** [comment]
@robodoo rebase-ff
